Foundation/NSSwiftRuntime.swift

```diff
@@ -14,10 +14,16 @@ import CoreFoundation
 // This mimics the behavior of the swift sdk overlay on Darwin
 #if os(OSX) || os(iOS)
 @_exported import Darwin
-#elseif os(Linux)
+#elseif os(Linux) //// || os(Android) // Causes ambiguity in NSScanner.swift
```

Comment:
I should rephrase: if the code had no other issues, it would be fine where it is, but it cannot be a public API because there is a public API in the stdlib that provides the same facility.

In fact, the code in NSScanner is there because it pre-dates the revised FloatingPoint protocols and works around a common limitation of floating point protocols prior to their current state. The internal protocol _FloatLike could actually be ripped out entirely and replaced with the new FloatingPoint, and probably should be if it's causing you grief here. That would be a rather straightforward thing to implement.